RentACar/RentACar/CRUD/CarCRUD.cs

```diff
@@ -1,69 +1,57 @@
-﻿using System;
-using System.IO;
-using System.Text.Json;
+﻿using System.IO;
+using RentACar.Utilities;
 using System.Collections.Generic;
 using System.Linq;
-
+using RentACar.CRUD;
 
 namespace RentACar
 {
-    public class CarCRUD
+    public class CarCRUD : ICarCRUD
     {
-        private readonly string _jsonFile = @"..\..\..\CRUD\Cars.json";
-
-        private List<Car> DeserializeJsonToList(string jsonFile)
-        {
-            string existingJsonToString = File.ReadAllText(jsonFile);
-            List<Car> deserializedJson = JsonSerializer.Deserialize<List<Car>>(existingJsonToString);
-
-            return deserializedJson;
-        }
-
-        private void SerializeListToJson(List<Car> cars)
-        {
-            var options = new JsonSerializerOptions { WriteIndented = true };
-
-            string listToString = JsonSerializer.Serialize(cars, options);
-            File.WriteAllText(_jsonFile, listToString);
-        }
+        private readonly string _jsonFile = @"..\..\..\Data\Cars.json";
 
         public Car Create(Car car)
         {
-            var cars = new List<Car>();
+            var cars = (File.Exists(_jsonFile)) ? CrudHelper.DeserializeJsonToList(_jsonFile) : new List<Car>();
 
-            if (File.Exists(_jsonFile))
-            {
-                cars = DeserializeJsonToList(_jsonFile);
+            car.Id = CrudHelper.GetNewId(cars);
+            cars.Add(car);
 
-                car.Id = cars.Last().Id + 1;
-                cars.Add(car);
-            }
-            else
-            {
-                car.Id = 0;
-                cars.Add(car);
-            }
-
-            SerializeListToJson(cars);
+            CrudHelper.SerializeListToJson(cars, _jsonFile);
 
             return car;
         }
 
         public Car Get(int id)
         {
-            // Code here
-            return new Car();
+            if (!File.Exists(_jsonFile)) return null;
+
+            var cars = CrudHelper.DeserializeJsonToList(_jsonFile);
+
+            var car = cars.Where(e => e.Id == id).FirstOrDefault();
+
+            return car ?? null;
```

Comment:
just being curious: what made you put this line here?

Reply:
It was a slip in my mind! 🤣 I thought "Oh, if the car is null, this method has to return null"; buy yes, in that case, it will already return null! Thanks for noticing!
I was thinking, if i change "cars.Where(e => e.Id == id).FirstOrDefault();" by just "cars.FirstOrDefault(e => e.Id == id));" it will be more readable and perform a little better, right?

Reply:
not sure about performance (not noticeable for sure)
not strongly opinionated about one form or the other, along as you are consistent through all the app then I don't see a problem